### ISSUE TITLE
Are allotropes of carbon hydrocarbons?

### ISSUE THREAD

**Issue by MadeleineBirchfield** [UniMath/agda-unimath#232]
There are certain molecules called allotropes of carbon which consist entirely of carbon atoms. Examples include carbon nanotubes and fullerines. 
https://en.wikipedia.org/wiki/Allotropes_of_carbon
https://en.wikipedia.org/wiki/Carbon_nanotube
https://en.wikipedia.org/wiki/Fullerene
With the current definition of hydrocarbon, these molecules are hydrocarbons with zero hydrogen atoms. 

But I'm not sure if the chemistry community would call allotropes of carbon 'hydrocarbons', common usage seems to imply that there is at least one hydrogen atom in a hydrocarbon.

**Comment by EgbertRijke** [comment]
I agree with you, hydrocarbons should probably have at least one hydrogen atom! Thank you again for your contribution!

**Comment by plt-amy** [comment]
I believe this is _already_ impossible. If we look at this confusing-perspective diagram of C₆₀ fullerene (from [PubChem](https://pubchem.ncbi.nlm.nih.gov/compound/Buckminsterfullerene), but I'll take full blame for the hand-drawn highlighting), we see that the highlighted carbon has 4 incident edges.

![image](https://user-images.githubusercontent.com/29957583/177427300-7aeccd44-cdd8-476b-a54a-4eb65886065b.png)

But this isn't allowed:

```
(c c' : vertex-Undirected-Graph-𝔽 G) → leq-ℕ
  (number-of-elements-is-finite (is-finite-type-𝔽 (pr2 G (standard-unordered-pair c c'))))
  3)
```

Translating, in the type of `hydrocarbon`s, each carbon (vertex in the underlying graph) can have at most three carbon-carbon bonds. This excludes every allotrope: they all have carbons making 4 carbon-carbon bonds.

**Comment by EgbertRijke** [comment]
The condition you mention here only excludes quadruple bonds between two carbon atoms in the molecule.

**Comment by MadeleineBirchfield** [comment]
According to Egbert Rijke in https://github.com/UniMath/agda-unimath/issues/234#issuecomment-1176187175 requiring the molecule to have at least one hydrogen atom implies that there are no quadruple bonds between two carbon atoms in the molecule, because the graph is connected.
